### ISSUE TITLE
[Feature Request] Remove netcore 3.1 support from MSAL 

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#3864]
Any public client features etc. with netcore3.1 explicit support can be removed, since netcore 3.1 is out of support as of December 13

**Comment by pmaytak** [comment]
Do you mean [Net Core 2.1](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/4e96195af87c624b450e65fefc29d3a5fd1877f9/src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj#L7)?

**Comment by bgavrilMS** [comment]
No. I mean netcoreapp 3.1 - from the MSAL.Desktop projects etc. 
I am not sure if we can remove netcoreapp2.1 from MSAL and rely on the netstandard tfm?

**Comment by TaviTruman** [comment]
`2>C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(1027,5): error NETSDK1149: Microsoft.Identity.Client.Desktop.dll cannot be referenced because it uses built-in support for WinRT, which is no longer supported in .NET 5 and higher.  An updated version of the component supporting .NET 5 is needed. For more information, see https://aka.ms/netsdk1149
`
Need to remove netcore3.2 support because it blocks usage of the Windows-Toolkit-Graph-Controls in WPF that target .NET 7

**Comment by pmaytak** [comment]
@TaviTruman For Net 7, your app needs to target at least net7.0-windows10.0.17763.0 and **not** reference Microsoft.Identity.Client.Desktop. It needs to be referenced only when targeting Net Framework or Net Core.
